### DISCUSSION TITLE
execute tsx files

### DISCUSSION THREAD

**Discussion by arashi-dev** [johnlindquist/kit#1348] [Q&A]
Is it possible to have a task with `.tsx` extension? if yes, how? if not, where should I put my jsx codes? inside `/lib` folder?

**Comment by johnlindquist** [comment]
No, what's your use-case? Curious to see what you're expecting to work

**Reply by arashi-dev** [reply]
I would like to use react components to display results after my script runs. for now, I have succeeded in having my JSX codes in `.tsx` files inside `/lib` folder and converted them into strings using `renderToString()` function, and imported the result in my main script file. however, it would be great if there was a cleaner way to do that. like being possible to have a folder structure inside `scripts` folder and also be able to execute `.jsx|.tsx` extensions (or even maybe support `react` under the hood which would be great)

**Reply by johnlindquist** [reply]
I'll add it to the roadmap to see what's possible with the on-the-fly compiling.

**Reply by arashi-dev** [reply]
Good to hear that! thanks
